addons/mrp/models/mrp_workorder.py

```diff
@@ -416,7 +416,7 @@ def _onchange_operation_id(self):
 
     @api.onchange('date_planned_start', 'duration_expected', 'workcenter_id')
     def _onchange_date_planned_start(self):
-        if self.date_planned_start and self.duration_expected and self.workcenter_id:
+        if self.date_planned_start and self.duration_expected is not False and self.workcenter_id:
```

Comment:
It's possible to manually delete the `planned_start` field (i.e. set it to False) and/or have it False in other cases have a similar error occur. It's a little hard to reproduce though, here's one case where you will have another boolean comparison error (when your fix is applied):

- create a new MO from scratch (no BoM)
- add 2 (workorder) operations, the 2nd one has a Scheduled Date set, but the 1st one does not + save
- confirm => try to add a Scheduled Start Date for the 1st line (i.e. the one without one) => error

Here's a video of it https://drive.google.com/file/d/1EflKAnXtFSCSX9C6jCq5q-99QmaGoSDN/view?usp=drivesdk

This may be due to another date being False when it's expected to not be false though. That can maybe be another PR, but if the issue reason is similar enough maybe this PR can make sure  `_plan_workorder` and `_plan_workorders` are more resistant to missing date values?